mautic/mautic
# Fix undefined variable while creating form with API

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ y ]
| New feature/enhancement? (use the a.x branch)      | [ n ]
| Deprecations?                          | [ n ]
| BC breaks? (use the c.x branch)        | [ n ]
| Automated tests included?              | [ n ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
I've got error with missing variable while creating form via API with checkbox after upgrading to version 5 alpha.

```
{
    "errors": [
        {
            "message": "Variable \"list\" does not exist.",
            "code": 500,
            "type": null
        }
    ],
    "trace": [
        {
            "namespace": "",
            "short_class": "",
            "class": "",
            "type": "",
            "function": "",
            "file": "/var/www/html/app/bundles/FormBundle/Resources/views/Field/group.html.twig",
            "line": 229,
            "args": []
        },
....
```



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Send POST API request to create form with payload:
```
{
  "name": "Test Form5",
  "formType": "standalone",
  "description": "API test",
  "fields": [
    {
      "label": "email",
      "alias": "email",
      "type": "email",
      "leadField": "email"
    },
    {
      "label": "checkbox",
      "alias": "checkbox",
      "type": "checkboxgrp",
      "properties": {
        "optionlist": {
          "list": [
            {
              "label": "val1",
              "value": "val1"
            }
          ]
        }
      }
    },
    {
      "label": "Submit",
      "alias": "submit",
      "type": "button"
    }
  ]
}
```
3. Response should have HTTP code 201 Created
